src/assets/TokenRatesController.ts

```diff
@@ -181,6 +192,10 @@ export class TokenRatesController extends BaseController<
         timestamp: 0,
         data: null,
       },
+      supportedVsCurrencies: {
```

Comment:
Should this be on the controller state? :thinking: I'm not sure anyone else has use for this data. Internal data doesn't need to be in the controller state. Anything in state gets synced with the UI, so we don't want to include more than we need to.

Reply:
@wachunei I think this also applies to the supportedChains state that you added? Do you object to my making it a private variable on the controller instead of controller state?

Reply:
@adonesky1  I added those to the state because as far as I know only the state gets persisted on the products, so the cache works as intended, is this correct?

Reply:
Sounds good, now I'm just wondering how long the memory is held on the mobile platforms when the app is sent to the background and brought back to the foreground. Has this been researched? 🤔

Reply:
Not that I'm aware of but I can ask some folks on the mobile team if they have any idea.

Reply:
That is a great question

Reply:
@Gudahtt @wachunei per the feedback in this [thread](https://consensys.slack.com/archives/GBW7S9FSN/p1632329799275800) it seems we can move these to class variables instead of controller state.

Reply:
🆗, If mobile team is available once they do upgrade to this version we could try a testflight release just to confirm the behavior 👍